### DISCUSSION TITLE
How to distinguish different boards when USB serial numbers are all "123456"?

### DISCUSSION THREAD

**Discussion by bikeNomad** [micropython/micropython#12864] [ESP32] [answered]
I have several ESP32-S3 boards and can't distinguish them when they're plugged into my computer because they all show the same USB serial number (123456). Is there some way to use the unique chip ID as a basis for the USB serial number?

I'm using the latest Micropython build.

**Comment by jimmo** [comment]
The Espressif IDF sets this value to `CONFIG_TINYUSB_DESC_SERIAL_STRING` (which defaults to "123456" as you've noticed). The IDF doesn't seem to have a simple way to make this automatically use the chip ID.

So you can easily override this value (in sdkconfig) but not easily make it automatic. To do that means we need to provide our own implementation of the descriptor configuration (replacing the IDF's Kconfig-based implementation), and pass this to `tinyusb_driver_install` in esp32/usb.c.

Fortunately we already have such an implementation for the other tinyusb ports (rp2, samd, etc, see shared/tinyusb/mp_usbd_descriptor.c), but adapting it to esp32 is not completely trivial.

**Comment by karlp** [comment, marked as answer]
This is fixed automatically in 1.24 (via 4247921c4e), if you set CONFIG_TINYUSB_DESC_SERIAL_STRING to ""
